### PR TITLE
Add support for V3 Roles

### LINKED ISSUE #1078: Missing V3 Roles API
Missing implementation of http://v3-apidocs.cloudfoundry.org/version/3.92.0/#roles

### PR DESCRIPTION
**Pull request by radito3**
Fixes #1078

### CODE REVIEW

**Comment by twoseat** [comment]
Thanks for the PR @radito3! I did a quick check through the code and the only thing that jumped out at me is we're naming all new V3 endpoints with their version, e.g. RolesV3. If you could update that, and sign the EasyCLA, I should be able to get this merged quickly.

**Comment by radito3** [comment]
@twoseat Thanks for the comment! I renamed the appropriate classes from `Roles` to `RolesV3` and signed the CLA. Could you take another look at the pr?

File: cloudfoundry-client/src/test/java/org/cloudfoundry/client/v3/roles/CreateRoleRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * limitations under the License.
+ */
+
+package org.cloudfoundry.client.v3.roles;
+
+import org.cloudfoundry.client.v3.Relationship;
+import org.cloudfoundry.client.v3.ToOneRelationship;
+import org.junit.Test;
+
+public class CreateRoleRequestTest {
```
Comment (twoseat) [inline cloudfoundry-client/src/test/java/org/cloudfoundry/client/v3/roles/CreateRoleRequestTest.java:23]: Please add a couple of tests for the @Value.Check in the request.

File: cloudfoundry-operations/src/test/java/org/cloudfoundry/operations/applications/ApplicationManifestUtilsTest.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 import static org.cloudfoundry.operations.applications.ApplicationHealthCheck.PORT;
 import static org.junit.Assume.assumeTrue;
 
+@org.junit.Ignore
```
Comment (twoseat) [inline cloudfoundry-operations/src/test/java/org/cloudfoundry/operations/applications/ApplicationManifestUtilsTest.java:36]: Should this be here?

**Review by twoseat — CHANGES_REQUESTED** [review]
Looks great with just a couple of issues. Once updated I'll put some integration tests behind it, which almost always reveal a problem (generally a gap between the API and the documentation, not us!)

File: cloudfoundry-operations/src/test/java/org/cloudfoundry/operations/applications/ApplicationManifestUtilsTest.java (hunk 2 above)
Comment (radito3) [reply cloudfoundry-operations/src/test/java/org/cloudfoundry/operations/applications/ApplicationManifestUtilsTest.java:36]: Oops, my bad. Will remove it

File: cloudfoundry-client/src/test/java/org/cloudfoundry/client/v3/roles/CreateRoleRequestTest.java (hunk 1 above)
Comment (radito3) [reply cloudfoundry-client/src/test/java/org/cloudfoundry/client/v3/roles/CreateRoleRequestTest.java:23]: Will do

**Comment by radito3** [comment]
@twoseat  Could you take another look?

**Pull request closed without merging**